Core/src/Geometry/CuboidVolumeBuilder.cpp

```diff
@@ -114,6 +123,22 @@ std::pair<double, double> Acts::CuboidVolumeBuilder::binningRange(
       minMax.second = surfacePosMax;
     }
   }
+
+  //Use the volume boundaries as limits for the binning
+  //and check that the layers are inside the defined tracking volume
+    
+  if (minMax.first >= min_volume_boundaries(binX))
+    minMax.first = min_volume_boundaries(binX);
```

Comment:
How about 

```cpp
minMax.first = std::min(minMax.first, minVolumeBoundaries(binX);
minMax.second = std::max(minMax.second, maxVolumeBoundaries(binX);
```

? This should do the same thing I believe.

Reply:
Done